### PR TITLE
Amend breaking change in RecurringHostedServiceBase

### PR DESCRIPTION
**Pull request by Zeegaan**
Amend breaking change

### CODE REVIEW

File: src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs [hunk 1]
```diff
@@ -41,6 +43,13 @@ protected RecurringHostedServiceBase(ILogger logger, TimeSpan period, TimeSpan d
             _delay = delay;
         }
 
+        // Scheduled for removal in V11
+        [Obsolete("Please use constructor that takes an ILogger instead")]
+        protected RecurringHostedServiceBase(TimeSpan period, TimeSpan delay)
+        : this(StaticServiceProvider.Instance.GetRequiredService<ILogger>(), period, delay)
```
Comment (p-m-j) [inline src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs:49]:
```suggestion
        : this(StaticApplicationLogging.CreateLogger<RecurringHostedServiceBase>(), period, delay)
```

File: src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs [hunk 2]
```diff
@@ -34,13 +36,20 @@ public abstract class RecurringHostedServiceBase : IHostedService, IDisposable
...
         {
             _logger = logger;
             _period = period;
             _delay = delay;
         }
 
+        // Scheduled for removal in V11
+        [Obsolete("Please use constructor that takes an ILogger instead")]
+        protected RecurringHostedServiceBase(TimeSpan period, TimeSpan delay)
+        : this(StaticServiceProvider.Instance.GetRequiredService<ILoggerFactory>().CreateLogger<RecurringHostedServiceBase>(), period, delay)
```
Comment (p-m-j) [inline src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs:49]:
```suggestion
        : this(StaticApplicationLogging.CreateLogger<RecurringHostedServiceBase>(), period, delay)
```

File: src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs [hunk 3]
```diff
@@ -22,7 +23,7 @@ public abstract class RecurringHostedServiceBase : IHostedService, IDisposable
         /// </summary>
         protected static readonly TimeSpan DefaultDelay = TimeSpan.FromMinutes(3);
 
-        private readonly ILogger _logger;
+        private readonly ILogger<RecurringHostedServiceBase> _logger;
```
Comment (p-m-j) [inline src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs:26]:
```suggestion
        private readonly ILogger _logger;
```

**Review by p-m-j — APPROVED** [review]
~LGTM~

**Comment by p-m-j** [comment]
LGTM!

File: src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs [hunk 4]
```diff
@@ -71,7 +81,7 @@ public async void ExecuteAsync(object state)
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex, "Unhandled exception in recurring hosted service {serviceName}.", GetType().Name);
+                _logger.LogError(ex, "Unhandled exception in recurring hosted service {serviceName}.");
```
Comment (vsilvar) [inline src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs:84]: Noticed you removed the type parameter but missed the placeholder `{serviceName}` in the log message itself.

Comment (Zeegaan) [reply src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs:85]: @vsilvar good catch, we've toyed a bit around with different loggers so this was just an accidental removal 😅

File: src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs (hunk 1 above)
Comment (p-m-j) [reply src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs]: was bad idea as the hosted services startup before the generic web host starts up (which inits static application logging).

**Comment by Zeegaan** [comment]
Cherry picked for 9.4

**Pull request merged**